### PR TITLE
Add Open Liberty 21.0.0.12 and 22.0.0.12

### PR DESCRIPTION
**Pull request by fmhwong**
Add newer Open Liberty images 21.0.0.12 and 22.0.0.12.

### CODE REVIEW

File: smoke-tests/images/servlet/build.gradle.kts [hunk 1]
```diff
@@ -36,6 +36,10 @@ val targets = mapOf(
     // running configure.sh is failing while building the image with Java 19
     ImageTarget(listOf("21.0.0.10"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
     ImageTarget(listOf("21.0.0.10"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
```
Comment (trask) [inline smoke-tests/images/servlet/build.gradle.kts:38]: I think we can drop this in favor of `21.0.0.12`

```suggestion
```

Comment (fmhwong) [reply smoke-tests/images/servlet/build.gradle.kts:38]: Should I remove them in a follow up PR after we switched to the new images? If I remove them now, will it break existing tests?

Comment (trask) [reply smoke-tests/images/servlet/build.gradle.kts:38]: it's confusing, but it won't break tests

the tests will continue using the old tag until you update the tag in #7857:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8a7e8a8ca203a3ffbfd4baeb322213f600bc05de/smoke-tests/src/test/groovy/io/opentelemetry/smoketest/AppServerTest.groovy#L57

Comment (laurit) [reply smoke-tests/images/servlet/build.gradle.kts:38]: It is ok to remove them now. Tests use images with a specific tag, they won't automatically switch to the new images.

File: smoke-tests/images/servlet/build.gradle.kts [hunk 2]
```diff
@@ -34,8 +34,6 @@ val targets = mapOf(
   "liberty" to listOf(
     ImageTarget(listOf("20.0.0.12"), listOf("hotspot", "openj9"), listOf("8", "11"), mapOf("release" to "2020-11-11_0736")),
     // running configure.sh is failing while building the image with Java 19
```
Comment (trask) [inline smoke-tests/images/servlet/build.gradle.kts:36]: @fmhwong do you know if this is still an issue, or can we (should we?) test with Java 19?

Comment (fmhwong) [reply smoke-tests/images/servlet/build.gradle.kts:36]: Java 19 should be supported starting with version 22.0.0.10.

File: smoke-tests/images/servlet/build.gradle.kts [hunk 3]
```diff
@@ -34,8 +34,10 @@ val targets = mapOf(
...
-    ImageTarget(listOf("21.0.0.10"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
-    ImageTarget(listOf("21.0.0.10"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
+    ImageTarget(listOf("21.0.0.12"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "2021-11-17_1256")),
+    ImageTarget(listOf("21.0.0.12"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "2021-11-17_1256")),
+    ImageTarget(listOf("22.0.0.12"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "22.0.0.12")),
+    ImageTarget(listOf("22.0.0.12"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "22.0.0.12")),
```
Comment (trask) [inline smoke-tests/images/servlet/build.gradle.kts:40]:
```suggestion
    ImageTarget(listOf("22.0.0.12"), listOf("hotspot"), listOf("8", "11", "17", "19"), mapOf("release" to "22.0.0.12")),
    ImageTarget(listOf("22.0.0.12"), listOf("openj9"), listOf("8", "11", "17", "19"), mapOf("release" to "22.0.0.12")),
```

File: smoke-tests/images/servlet/build.gradle.kts [hunk 4]
```diff
@@ -34,8 +34,10 @@ val targets = mapOf(
   "liberty" to listOf(
     ImageTarget(listOf("20.0.0.12"), listOf("hotspot", "openj9"), listOf("8", "11"), mapOf("release" to "2020-11-11_0736")),
     // running configure.sh is failing while building the image with Java 19
```
Comment (trask) [inline smoke-tests/images/servlet/build.gradle.kts:36]:
```suggestion
    // Java 19 is not supported until 22.0.0.10
```

**Review by trask — APPROVED** [review]

File: smoke-tests/images/servlet/build.gradle.kts [hunk 5]
```diff
@@ -33,9 +33,11 @@ val targets = mapOf(
...
   "liberty" to listOf(
     ImageTarget(listOf("20.0.0.12"), listOf("hotspot", "openj9"), listOf("8", "11"), mapOf("release" to "2020-11-11_0736")),
-    // running configure.sh is failing while building the image with Java 19
-    ImageTarget(listOf("21.0.0.10"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
-    ImageTarget(listOf("21.0.0.10"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "2021-09-20_1900")),
+    ImageTarget(listOf("21.0.0.12"), listOf("hotspot"), listOf("8", "11", "17"), mapOf("release" to "2021-11-17_1256")),
+    ImageTarget(listOf("21.0.0.12"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "2021-11-17_1256")),
+    // Java 19 is not supported until 22.0.0.10
+    ImageTarget(listOf("22.0.0.12"), listOf("hotspot"), listOf("8", "11", "17", "19"), mapOf("release" to "22.0.0.12")),
+    ImageTarget(listOf("22.0.0.12"), listOf("openj9"), listOf("8", "11", "17", "19"), mapOf("release" to "22.0.0.12")),
```
Comment (trask) [inline smoke-tests/images/servlet/build.gradle.kts:40]: build is failing with

```
Could not build image: manifest for ibm-semeru-runtimes:open-19-jdk not found: manifest unknown: manifest unknown
```

```suggestion
    ImageTarget(listOf("22.0.0.12"), listOf("openj9"), listOf("8", "11", "17"), mapOf("release" to "22.0.0.12")),
```

Comment (fmhwong) [reply smoke-tests/images/servlet/build.gradle.kts:40]: Just realized Semeru runtime doesn't have Java 19 yet.

**Comment by trask** [comment]
@fmhwong I just merged, when this build finished, the new images should be available: https://github.com/open-telemetry/opentelemetry-java-instrumentation/actions/runs/4247218894

**Pull request merged**